tls/data_source_public_key.go

```diff
@@ -0,0 +1,75 @@
+package tls
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourcePublicKey() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourcePublicKeyRead,
+		Schema: map[string]*schema.Schema{
+			"private_key": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "PEM formatted string to use as the private key",
+			},
+
+			"private_key_path": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "File path of the PEM formatted string to use as the private key",
+				ForceNew:    true,
```

Comment:
`ForceNew` is not needed in data sources. This generally controls deletion/creation workflows, both of which data sources lack.